---
title: "Data plots and checks"
author: "Gordana Popovic"
date: "21/05/2020"
output: html_document
---

The first line is saying "don't show this setup instructions in the final report"
The second line is saying "but for all other parts, show the code chunks in the final report":

```{r setup, include=FALSE} 
knitr::opts_chunk$set(echo = TRUE)
```

The first says "dont show messages or warnings in the document"
loads the tidyverse and readxl packages which are used to manipulate and view data. Readxl helps read excel files in R:


```{r message=FALSE, warning=FALSE}
library(tidyverse)
library(readxl)
```

This code reads excel files and stores their contents under the names ovarian and coefs:
na = c("", ".", "Missing"))

```{r}
ovarian <-read.csv("../data/normalized_data.csv",na = c("NA"))
coefs<-read.csv("../data/coefs.csv")
```


Renames columnes, 
Changes unknown values to NA, 
Creates age categories, 
Creates stage categories, 
Shortens an ID

(changed column titles in excel file, changed "MOCOG ID" to "MOCOG.ID", code now running). 
Patients under 53 are assigned a zero and are in the reference group according to Millstein et al., paper.

```{r}
Ovarian <- ovarian %>% 
  mutate(#OTTA_ID = `MOCOG.ID`,
         refage_revised = AgeAtDiagnosis,
         #stagenew = ifelse(stagenew == "unknown", NA,stagenew),
         age.fq2 = ifelse(refage_revised>=53&refage_revised<60,1,0),#recode age
         age.fq3 = ifelse(refage_revised>=60&refage_revised<67,1,0),
         age.fq4 = ifelse(refage_revised>=67,1,0),
         stage.f1 = ifelse(stagenew%in%c(1,2),1,0), #recode stage
         stage.f8 = ifelse(is.na(stagenew),1,0)) %>% 
  mutate(patientIDi = substr(OTTA_ID,23,35))

```

This code created a summary for each group. 
It looks at the minimum and maximum age in each group. 
It also looks at the number of rows (observations) in each group.
It looks at the number of people in each stage. 

(write csv not working, I think age object has not been created?)

```{r}

#check age recoding
Ovarian%>% 
  group_by(age.fq2,age.fq3,age.fq4) %>% 
  summarise(min_age=min(refage_revised),max_age=max(refage_revised),N=length(refage_revised))


#Check stage recoding
Ovarian%>% 
  group_by(stage.f1,stage.f8) %>% 
  summarise(stageI=sum(stagenew==1),stageII=sum(stagenew==2),not_stage=sum(is.na(stagenew)),
            N=length(stagenew))

#write.csv(age,file="../results/age_code.csv")



```

If comments are removed, the code groups the data into two columns, pre_treatment_primary and tissue_site. It counts the number of observations in each group.

(changed "patient_primary_treatment" to "pre_treatment_primary")

```{r}
Ovarian %>% 
group_by(pre_treatment_primary,tissue_site) %>% 
count()

```


Renames genes so that they are consistent between the two files. 
Matches gene names to column names, identifies which genes are in both files.
Find missing genes
Identifies matching genes. 
Retrieves coefficientss for unmatches genes. 
Calculates the mean absolute efficient. 

```{r}

coefs$Gene[coefs$Gene=="PD.1"]="PD_1" #change gene name in coefs to match data
col_id=match(coefs$Gene, colnames(Ovarian)) 
coefs$Gene[is.na(col_id)]
coefs$Coefficient[is.na(col_id)]
mean(abs(coefs$Coefficient))

```

Reorder genes in data to match order in coefs

```{r}
coefs_sub=coefs[!coefs$Gene%in%coefs$Gene[is.na(col_id)],]
Ovarian_ord = Ovarian[,match(coefs_sub$Gene, colnames(Ovarian))]
all(colnames(Ovarian_ord)==coefs_sub$Gene)

```

#predict
This code creates a matrix (each row is a patient and each column is gene expression level). Beta takes the coefs and turns them into a column of numbers. Each number corresponds to a gene.
The gene expression level for each patient is multiplied by the weights, then adds them together to get a predicted score for each patient. 
New column is created for these results called "predict".


```{r}
X=as.matrix(Ovarian_ord)
beta=as.matrix(coefs_sub$Coefficient)
Ovarian$predict=X%*%beta #matrix multiplication 


```

This code creates a new column that groups patients into quintiles based on their prediction score. 

```{r}
Ovarian<-Ovarian %>% 
  mutate(quintile=cut(predict,c(-100,-0.86697992, -0.37336052, -0.07486426,  0.20987383,100), 
                      labels=paste0("C",c(1:5))))
```

The file is then saved. 

```{r}
write.csv(Ovarian,file="../results/ascites_predictions.csv") 
```

Comparing predictions with Josh's file 
We used quintile - predict
Josh used signature.quintile - signature

```{r,eval=FALSE}

#all.equal(Ovarian$predict, Ovarian$signature)
Ovarian%>%
  select(OTTA_ID,stagenew, predict, signature)%>%
  mutate(difference= predict - signature)%>%
  ggplot(aes(OTTA_ID, difference)) + 
  geom_point()


```

```{r, eval=FALSE}
table(Ovarian$signature.quintile,Ovarian$quintile)
```

Compare SPOT scores between tissue sites. 

```{r}
#Reorganise data to split different sites for each patient. 
Ovarian_split <- Ovarian %>% 
  separate(patientIDi, into = c("PatientID", "Site"), sep = "\\^", remove = FALSE)



```























